Updates Portilla-Simoncelli synthesis in docs/tests (#381)
plenoptic-org/plenoptic: docs/tutorials/models/portilla_simoncelli/ps_optimization.md

    import os
    os.environ["OMP_NUM_THREADS"] = 1
    ```

In practice, you are encouraged to experiment with the two above settings to find the best configuration for your problem.

(ps-loss-function)=
## Custom loss function

When using the LBFGS optimizer configured as described above, plenoptic's Portilla-Simoncelli metamer synthesis has a lower loss in every component than the MATLAB implementation, with the notable exception of the variance of highpass residuals. The two synthesis implementations (as discussed in [](ps-mat-diffs)) are very different from each other: the MATLAB code does something akin to coordinate descent, where it optimizes each set of statistics separately, whereas the plenoptic code optimizes all of them together by trying to minimize the overall loss. This results in plenoptic weighting each statistic approximately equally and thus, implicitly, being considered equally important. They thus each have an approximately equal error at any given moment during synthesis, whereas the MATLAB code drops the error in the variance of the highpass residuals *incredibly* rapidly, matching it with much higher precision than the other statistics.

> In order to make plenoptic's synthesis perform similarly, we need to make this statistic more important than the others. We do this by using a custom loss function which reweights the representation of each image before computing the L2-norm of the difference these representations. We do this using the {func}`~plenoptic.tools.optim.portilla_simoncelli_loss_factory` function, which takes in an instance of the {class}`~plenoptic.simulate.models.portilla_simoncelli.PortillaSimoncelli` model and an image, and returns a loss function. (This is necessary because the shape of the model output depends on the parameters used to initialize it.)

Reviewer: ```suggestion
In order to make plenoptic's synthesis perform similarly, we need to make this statistic more important than the others. We do this by using a custom loss function which reweights the representation of each image before computing the L2-norm of the difference these representations, see `~plenoptic.tools.optim.portilla_simoncelli_loss_factory` for more details. 
```

I would trim this down
